### PR TITLE
add role and collection in root

### PR DESCRIPTION
**Pull request by sbaerlocher**
##### SUMMARY
With this function it is now possible to place the requirements.yml with Roles and Collections in the root directory of the Playbook. So it is anaglog to different programming languages which have the dependency file in the root directory. 

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME

 - Playbooks

### CODE REVIEW

**Comment by rebeccahhh** [comment]
@sbaerlocher we will probably wait for #8312 is merged (as it is quite close) to look at your PR.
Congrats on your first contribution to the AWX project. :confetti_ball: 
cc: @AlanCoding @elyezer @thedoubl3j

File: awx/playbooks/project_update.yml [hunk 1]
```diff
@@ -211,3 +211,47 @@
...
+    - block:
+        - name: detect requirements.(yml/yaml)
+          stat:
+            path: "{{project_path|quote}}/requirements{{ item.ext }}"
+          with_items: "{{ yaml_exts }}"
+          register: doesRequirementsExist
+
+        - name: fetch galaxy roles from requirements.(yml/yaml)
+          command: >
+            ansible-galaxy role install -r {{ item.path }}
```
Comment (thedoubl3j) [inline awx/playbooks/project_update.yml]: `item.stat.path` is what I had to change to. did some testing and `path` isn't available to the loop but we can steal the path from stat.

File: awx/playbooks/project_update.yml [hunk 2]
```diff
@@ -211,3 +211,47 @@
...
+          when: doesRequirementsExist.stat.exists and collections_enabled|bool
+          changed_when: "'Installing ' in galaxy_collection_result.stdout"
+          environment:
+            ANSIBLE_FORCE_COLOR: false
+            ANSIBLE_COLLECTIONS_PATHS: "{{projects_root}}/.__awx_cache/{{local_path}}/stage/requirements_collections"
+            GIT_SSH_COMMAND: "ssh -o StrictHostKeyChecking=no"
+
+      when:
+        - "ansible_version.full is version_compare('2.9', '>=')"
+      tags:
```
Comment (thedoubl3j) [inline awx/playbooks/project_update.yml:255]: might want to move these around to under each task specifically but once we the other thing in, we can match that pattern.

File: awx/playbooks/project_update.yml [hunk 3]
```diff
@@ -211,3 +211,47 @@
...
+          with_items: "{{ doesRequirementsExist.results }}"
+          when: doesRequirementsExist.stat.exists and roles_enabled|bool
+          changed_when: "'was installed successfully' in galaxy_result.stdout"
+          environment:
+            ANSIBLE_FORCE_COLOR: false
+            GIT_SSH_COMMAND: "ssh -o StrictHostKeyChecking=no"
+
+        - name: fetch galaxy collections from requirements.(yml/yaml)
+          command: >
+            ansible-galaxy collection install -r {{ item.path }}
```
Comment (thedoubl3j) [inline awx/playbooks/project_update.yml]: same as the other

File: awx/playbooks/project_update.yml (hunk 2 above)
Comment (thedoubl3j) [reply awx/playbooks/project_update.yml:255]: these being the tags, miss clicked that. apologies

**Comment by rebeccahhh** [comment]
@sbaerlocher the PR this one was waiting on is merged

**Comment by sbaerlocher** [comment]
@rebeccahhh I will make the adjustments tomorrow.

**Comment by rebeccahhh** [comment]
@sbaerlocher no rush, I'll take a look whenever you get to it, just tag me if I'm not being fast enough :)

**Comment by ryanpetrello** [comment]
cc @sbaerlocher @rebeccahhh @AlanCoding @thedoubl3j what's the state of this PR?

**Comment by thedoubl3j** [comment]
@ryanpetrello @sbaerlocher looks like at a minimum the original review I left needs to be added. Will default to @AlanCoding if he has differing thoughts. But as of now, the changing to `item.stat.path` will pattern match what is above it.

**Comment by rebeccahhh** [comment]
pending testing I'm good to merge it

**Comment by sbaerlocher** [comment]
Sorry for the long wait, I have now made the adjustments.

**Comment by AlanCoding** [comment]
You probably want to rebase for the playbook to make complete functional sense.

However, I have issues with the general approach. We are _constantly_ making heavy updates to this playbook, and any unnecessary duplication is going to make that maintenance unworkable. I left a comment in a prior PR here:

https://github.com/ansible/awx/pull/8312#issuecomment-703653506

In that case, I was uncomfortable with how it was written in 4 tasks, and was seeking to add a loop to consolidate it into 2 tasks. This current PR takes the 4 tasks and duplicates them to 8 tasks.

My suggestion here is double-down on the direction I was trying to go there:

```yaml
with_fileglob: "{{project_path|quote}}/(collections/)?requirements.ya?ml"
```

Using this, I'd expect the `project_update.yml` playbook to _lose_ lines on net, not add lines. It would still accomplish the goal here.

**Comment by AlanCoding** [comment]
Trying to move this forward, I put up https://github.com/ansible/awx/pull/9945

**Comment by AlanCoding** [comment]
Thanks for your contribution, I believe we will get this functionality from #9945, and I will work to move that forward.

**Pull request closed without merging**